typing/jkind.ml

```diff
@@ -1703,11 +1781,12 @@ let for_boxed_variant cstrs =
        (* CR layouts v2.8: This is sad, but I don't know how to account for
           existentials in the with_bounds. See doc named "Existential
           with_bounds". *)
-    then Builtin.value ~why:Boxed_variant
+    then Builtin.value ~why:Boxed_variant |> mark_best
```

Comment:
As discussed, I think we shouldn't `mark_best` here. GADTs will reliably have jkind `value` now, but this will change. Marking this `not_best` for now will, I think, make things easier later when we improve the jkind.